ofrak_core/ofrak/gui/server.py

```diff
@@ -739,6 +742,61 @@ async def search_for_vaddr(self, request: Request) -> Response:
         except NotFoundError:
             return json_response([])
 
+    @exceptions_to_http(SerializedError)
+    async def search_for_string(self, request: Request):
+        resource = await self._get_resource_for_request(request)
+        body = await request.json()
+        string_query_string = body["search_query"]
+        regex = body["regex"]
+        case_ignore = body["case_ignore"]
+        if not isinstance(string_query_string, str):
+            raise ValueError("Invalid search query.")
+        string_query: Union[bytes, re.Pattern[bytes]] = string_query_string.encode()
+        if case_ignore:
+            string_query = re.compile(string_query, re.IGNORECASE)
+        elif regex:
+            string_query = re.compile(string_query)
+
+        offsets = await resource.search_data(string_query)
+        if string_query == "":
+            return json_response(None)
+        found_resources = []
+        if len(offsets) > 0:
+            # found_resources[resource.get_id().hex()] = offsets
```

Comment:
Dead code? Also repeated below in 3 other spots.

```suggestion
```

Reply:
Ah yeah kind of, I'm planning on using those after ed gets his hex search in so it will automatically focus the location of the result in the hex view, but that should go in a seperate PR.